apps/root/src/pages/dca/create-position/components/step1/components/frequency-selector/index.tsx

```diff
@@ -47,29 +30,63 @@ const FrecuencySelector = ({ frequencies, handleFrequencyChange }: Props) => {
     trackEvent('DCA - Set frequency type', {});
   };
 
+  const validator = (nextValue: string) => {
+    // sanitize value
+    if (inputRegex.test(nextValue.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'))) {
+      handleFrequencyChange(nextValue);
+    }
+  };
+
+  const frequencyValueOptions = DCA_PREDEFINED_RANGES.map((range) => ({
+    value: range.value,
+    text: `${range.value} ${intl.formatMessage(
+      STRING_SWAP_INTERVALS[frequencyType.toString() as keyof typeof STRING_SWAP_INTERVALS].subject
+    )}`,
+  }));
+
+  const frequencyTypeOptions = frequencies.map((freq) => ({
+    value: freq.value,
+    text: capitalize(freq.label.adverb),
+  }));
+
   return (
-    <StyledFrequencyContainer>
-      <StyledFrequencyTypeContainer>
-        <Typography variant="body">
-          <FormattedMessage description="executes" defaultMessage="Executes" />
-        </Typography>
-        <FrequencyTypeInput options={frequencies} selected={frequencyType} onChange={onSetFrequencyType} />
-      </StyledFrequencyTypeContainer>
-      <StyledFrequencyValueContainer>
-        <Typography variant="body">
-          <FormattedMessage
-            description="howManyFreq"
-            defaultMessage="How many {type}?"
-            values={{
-              type: intl.formatMessage(
+    <>
+      <ContainerBox justifyContent="space-between">
+        <ContainerBox flexDirection="column" gap={3}>
+          <StyledDcaInputLabel>
+            <FormattedMessage description="investmentDuration" defaultMessage="Investment Duration" />
+          </StyledDcaInputLabel>
+          <ContainerBox gap={2}>
+            <TextField
+              id="investment-duration-input"
+              placeholder={`0 ${intl.formatMessage(
                 STRING_SWAP_INTERVALS[frequencyType.toString() as keyof typeof STRING_SWAP_INTERVALS].subject
-              ),
-            }}
-          />
-        </Typography>
-        <FrequencyInput id="frequency-value" value={frequencyValue} onChange={handleFrequencyChange} />
-      </StyledFrequencyValueContainer>
-    </StyledFrequencyContainer>
+              )}`}
+              onChange={(evt) => validator(evt.target.value.replace(/,/g, '.'))}
+              disabled={!from}
```

Comment:
why this?

Reply:
The onChange for this input has a if (!from) return. I wanted to keep this as it was! Are you suggesting to enable this input no matter the rest of the state?